qiskit/transpiler/passes/optimization/commutative_cancellation.py

```diff
@@ -70,12 +64,11 @@ def __init__(self, basis_gates=None, target=None):
 
         # build a commutation checker restricted to the gates we cancel -- the others we
         # do not have to investigate, which allows to save time
-        commutation_checker = CommutationChecker(
+        self.commutation_checker = CommutationChecker(
```

Comment:
Just so people don't assume this is publicly accessible:

```suggestion
        self._commutation_checker = CommutationChecker(
```